app/views/patients/pledge.js.erb

```diff
@@ -0,0 +1,16 @@
+$('.modal-content').html('')
+$('.modal').attr('id', 'pledge-modal')
+$('.modal-content').append(
+  '<%= escape_javascript(render partial: "patients/pledge", locals: { patient: @patient, disable_next: disable_continue?(@patient) })%>'
+);
+
+$('.modal').modal('toggle');
```

Comment:
quick clarification on what this set of javascript is doing? Worth rolling into a function?